modules/ballerina-core/src/test/java/org/ballerinalang/model/values/BMapValueTest.java

```diff
@@ -175,4 +177,23 @@ public void testBMapToString() {
         
         Assert.assertEquals(map.stringValue(), "{\"key1\":1, \"key2\":\"foo\", \"key3\":<bar>hello</bar>}");
     }
+
+    @Test
+    public  void testBMapOrder() {
+        BMap<String, BRefType> map = new BMap<>();
+        map.put(new String("Entry1"), new BString("foo"));
+        map.put(new String("Entry2"), new BString("bar"));
+        map.put(new String("Entry3"), new BString("foobar"));
+
+        Set set = map.keySet();
+        String[] ar = new String[3];
+        Iterator itr = set.iterator();
+        for (int i = 0; i < set.size(); i++) {
+         ar[i] = itr.next().toString();
+        }
+        Assert.assertEquals(map.get(ar[0]).stringValue(), "foo");
+        Assert.assertEquals(map.get(ar[1]).stringValue(), "bar");
+        Assert.assertEquals(map.get(ar[2]).stringValue(), "foobar");
+
+    }
```

Comment:
Shall we add a test case where the map is created and populated in ballerina level?